### DISCUSSION TITLE
Feature request: Bone rigging with inverse kinematics

### DISCUSSION THREAD

**Discussion by shimonshomron** [keiwando/toonsquid#69] [Ideas]
It would be great if bone rigging (both for raster and specially for vector points) would be available for the app.
It would be a DREAM - if those bones would have IK (Inverse kinematics) capabilities.
It would be DIVINE - if bones could control other bones (also known as smart bones) - this would allow complex animations/triggering and more.

The king of 2d bone rigging systems for reference is Moho Pro software which could be a great reference point.

**Comment by chriszesty** [comment]
I came here to ask for the same request. Having even just basic bone rigging would be a miles ahead of any other app out there!

**Reply by shimonshomron** [reply]
I found one app that has bone rigging for ipad, called- Animation pro. But no IK. 
It has quite wide variety of featured however it's pretty old and the interface logic (where every function is) is cumbersome and not intuitive, so I couldn't find myself investing the time to learn it throughly as it demands to take advantage of all it's features.
Toon squid regarding the interface (not just the looks but where everything is placed) is much more fluid and structured as other familiar apps, which makes it much more comfortable to get around in.
PLUS - it has base for vectors which has  huge potential for future features.

Indeed if proper bone system with IK  will be implemented it will have uniqueness on that sphere served on a suitable interface.

**Reply by kavlor** [reply]
@shimonshomron 
I use Anmation Pro quite a bit and have learnt it virtually inside out. It's quite a powerful app once you get to know it and surprisingly easy to set up rigs. It's no Moho for end results but IMO far more intuitive,quicker and easier to do animations. One of the great things about it is once you have a rig set up it isn't tied to the animations but can be dropped into any animation ,copy and pasted between frames and will still tween as long as it isn't pasted over a copy. You can even tween the keyframes as if they were individual frames (useful for rotoscoping). It also exports transparent video so you could load into Squid Toon.

**Reply by menannzoro** [reply]
This is the function I look forward to, because if this function is added, TS will be the best mobile animation application of all time
